xbmc/platform/posix/filesystem/SMBFile.cpp

```diff
@@ -34,9 +34,9 @@
 
 using namespace XFILE;
 
-void xb_smbc_log(const char* msg)
+void xb_smbc_log(void* private_ptr, int level, const char* msg)
 {
-  CLog::Log(LOGINFO, "{}{}", "smb: ", msg);
+  CLog::Log(LOGDEBUG, "smb: {}", msg);
```

Comment:
```suggestion
  const int logLevel = [level]()
  {
    switch (level)
    {
      case 0:
        return LOGWARNING;
      case 1:
        return LOGINFO;
      default:
        return LOGDEBUG;
    }
  }();
  CLog::Log(logLevel, "smb: {}", msg);
```
As `xb_smbc_log` will be called for critical errors and warnings even if component logging is disabled, it's probably a good idea to log these messages as such. I assume the `level` passed in here corresponds to the `-d|--debuglevel=DEBUGLEVEL` flag described in https://www.samba.org/samba/docs/current/man-html/smbclient.1.html.

Reply:
Nice idea. Let me test this.